sensor_core/dsp/dsp_manager.py

```diff
@@ -1,3 +1,15 @@
 import numpy as np
+import scipy.signal as signal
 
-#TODO: add modules for 1D dsp 
\ No newline at end of file
+#Moving average filter
+def moving_avg_filter(data, N):
+    data_arr = np.concatenate(([np.min(data)] * (N), data))
```

Comment:
instead of using np.min, maybe we should use the 10th percentile or something?

I could see a case where the minimum value is 0, but the most values are in the thousands and so we'd still be adding a ridiculously small value to an otherwise large dataset. 

Can you test the increase in computation time when using the 10th percentile value instead of np.min?